### ISSUE TITLE
Azure AD (AAD) Service Principal Support for GHEC EMUs (Preview)

### ISSUE THREAD

**Issue by github-product-roadmap** [github/roadmap#360]
### Summary

Many large GitHub Enterprise customers use [Service Principals](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals) to serve their automation needs in a centralised, auditable and secure fashion. Service Principals support the full identity and credential lifecycle for automation scenarios.  

### Intended Outcome

In this initiative, we will enable support for Azure AD Service Principles for GHEC EMUs as credentials on GitHub apps. We will later bring this capability to GHAE and GHES.

### How will it work?

To begin, enterprise customers will first need to configure a GHEC EMU Enterprise utilizing [Azure AD OpenID Connect (OIDC)](https://github.com/github/roadmap-internal/issues/414).  Next, an Azure AD admin will create the application and service principal objects in the application's AAD tenant.  Finally, an organization owner will register the Service Principal as a credential  on a GitHub Application within their GHEC EMU Enterprise, installed to one or more organizations.

**Comment by ankneis** [comment]
This issue is being closed as outdated. For more information, please check out this [Discussion post](https://github.com/github/roadmap/discussions/1014). Stay tuned for new additions to our refreshed public roadmap!

**Comment by ankneis** [comment]
We wanted to provide more details on why we removed this from the roadmap. Building and maintaining a strong platform-centric approach to integrations and automations is a key part of GitHub's product strategy going forward. Currently we are improving the [custom RBAC and least privilege permissions structures](https://github.com/github/roadmap/issues/1059) in GitHub Enterprise that are available to human and machine actors. As we have added support for more identity providers with the "Bring Your Own Identity Provider to GitHub" update, we will be evaluating how to apply the service principal platform goals more broadly to the GitHub service.

Feel free to upvote and share your thoughts in this [dedicated thread](https://github.com/orgs/community/discussions/45206) —your feedback really helps us prioritize!

If this thread doesn’t quite cover what you had in mind, feel free to start a new discussion.
